lib/beaker/host/unix/exec.rb

```diff
@@ -12,16 +12,17 @@ module Unix::Exec
   #
   # Will throw an exception RebootFailure if it fails
   def reboot(wait_time=10, max_connection_tries=9, uptime_retries=18)
+    require 'time'
+
     begin
-      original_uptime = exec(Beaker::Command.new("uptime"))
-      original_uptime_str = parse_uptime original_uptime.stdout
-      original_uptime_int = uptime_int original_uptime_str
+      original_boot_time_str = exec(Beaker::Command.new('who -b'), {:max_connection_tries => max_connection_tries, :silent => true}).stdout
+      original_boot_time_line = original_boot_time_str.lines.grep(/boot/).first
 
-      if self['platform'] =~ /solaris/
-        exec(Beaker::Command.new("reboot"), :expect_connection_failure => true)
-      else
-        exec(Beaker::Command.new("/sbin/shutdown -r now"), :expect_connection_failure => true)
-      end
+      raise Beaker::Host::RebootFailure, "Could not find system boot time using 'who -b': #{original_boot_time_str}" unless original_boot_time_line
+
+      original_boot_time = Time.parse(original_boot_time_line)
```

Comment:
we should handle the case where Time.parse fails and also raise a `Beaker::Host::RebootFailure`

Reply:
That's a good idea. It looks like it should [throw an `ArgumentError`](https://ruby-doc.org/stdlib-2.4.1/libdoc/time/rdoc/Time.html#method-c-parse).